### PR TITLE
feat(i18n): support observables and variables

### LINKED ISSUE #298: Pagination: Externalize hardcoded strings
## Detailed description

> Describe in detail the issue you're having.

There are untranslated strings in the Pagination template.

> Is this a feature request (new component, new icon), a bug, or a general issue?

Bug

> Is this issue related to a specific component?

Pagination

> What version of the Carbon Design System are you using?

2.0.7

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

ADI

### PR DESCRIPTION
**Pull request by cal-smith**
There's a bit to unpack here:

- `i18n.get` now optionally returns observables that update whenever `i18n.set` is called
- `i18n` exports a `replace` function that preforms variable substitution on the observable obtained from `i18n.get`
- better docs
- pagination now uses the update variable substitution (naming not finalized) (fixes #298 ... or at least that's the goal)
- i18n has useful tests now!

Pretty sure this hits all the important bits ... probably room for a few pipes and additional methods, but this is the base implementation

### CODE REVIEW

**Comment by zvonimirfras** [comment]
Maybe.

**Comment by stanislavgeorgiev** [comment]
Is there anything I can help with to make it go in faster?

File: src/i18n/i18n.service.ts [hunk 1]
```diff
@@ -1,16 +1,120 @@
...
+	 *
+	 * Returns the full translations object if path is not specified.
+	 *
+	 * @param path optional, looks like `"NOTIFICATION.CLOSE_BUTTON"`
+	 */
+	public get(path?) {
+		if (!path) {
+			return this.translationStrings;
+		}
+		const value = this.getValueFromPath(path);
```
[the remark this answers is not in the document]
Comment (cal-smith) [reply src/i18n/i18n.service.ts]: Ah yep, that's leftover from a previous implementation

[the remark this answers is not in the document]
Comment (cal-smith) [reply src/i18n/i18n.service.ts]: Actually I forgot, I'll add a comment, but that's there so we throw before we add the value to the translation map

[the remark this answers is not in the document]
Comment (cal-smith) [reply src/i18n/i18n.service.ts]: actually, even better, I'll just wrap it in a trycatch

**Pull request merged**